lib/puppet/ssl/inventory.rb

```diff
@@ -40,7 +44,10 @@ def rebuild
   def serials(name)
     return [] unless Puppet::FileSystem.exist?(@path)
 
-    File.readlines(@path).collect do |line|
+    # RFC 5280 says the cert subject may contain UTF8 - https://www.ietf.org/rfc/rfc5280.txt
+    # Note however that Puppet generated SSL files must only contain ASCII characters
+    # based on the validate_certname method of Puppet::SSL::Base
+    File.readlines(@path, :encoding => Encoding::UTF_8).collect do |line|
```

Comment:
https://tickets.puppetlabs.com/browse/PUP-7184 covers the addition of `readlines` to `Puppet::FileSystem`